pxd.c

```diff
@@ -181,8 +181,13 @@ static long pxd_ioctl_init(struct file *file, void __user *argp)
 	struct pxd_context *ctx = container_of(file->f_op, struct pxd_context, fops);
 	struct iov_iter iter;
 	struct iovec iov = {argp, sizeof(struct pxd_ioctl_init_args)};
+	int direction = WRITE;
 
-	iov_iter_init(&iter, READ, &iov, 1, sizeof(struct pxd_ioctl_init_args));
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,2,0)
+	direction = READ;
+#endif
+	  
+	iov_iter_init(&iter, direction, &iov, 1, sizeof(struct pxd_ioctl_init_args));
```

Comment:
direction is either ITER_SOURCE (for write) or ITER_DEST (for read).
Using READ or WRITE is incorrect here.